### PR TITLE
UITEN-34: Enable translation of headings on Settings

### PR DESCRIPTION
**Pull request by K-Felk**
The solution here is to provide translation strings for every plugin, 
and dynamically generate IDs  for them that can be passed to FormattedMessage.
the downside of this, of course, is that when we add new plugins, we have to 
remember to add new translation strings here, or they will just show up as 
IDs in the settings panel.  I can't see any better way to do it, though.

https://issues.folio.org/browse/UITEN-34

### CODE REVIEW

File: src/settings/SSOSettings/SamlForm.js [hunk 1]
```diff
@@ -89,6 +89,8 @@ class SamlForm extends React.Component {
       validateIdpUrl,
     } = this.props;
 
+    console.log(optionLists);
+
```
Comment (zburke) [inline src/settings/SSOSettings/SamlForm.js:93]: Whoops

File: translations/ui-tenant-settings/en.json [hunk 2]
```diff
@@ -121,6 +121,24 @@
   "settings.intervalPeriod.weeks": "Weeks",
   "settings.intervalPeriod.months": "Months",
 
+
+  "settings.pluginNames.create-inventory-records": "Create inventory records",
+  "settings.pluginNames.find-user": "Find User",
```
Comment (zburke) [inline translations/ui-tenant-settings/en.json:126]: FOLIO uses sentence case for everything, so for `find-user` use "Find user". For `find-eresource` use "Find e-resource" (singular). But acronyms (like FOLIO and PO) should be all-caps, so for `find-po-line` use "Find PO line", as you did for `find-erm...`. 

For the `finc-` plugins, what do you think about expanding the labels to "finance"?

**Review by zburke — CHANGES_REQUESTED** [review]
This looks great but there is a little bit of clean up work to do before you merge it. 

Also note that changes to locale files other than `en.json` will be overwritten on the next import from Lokalise so I typically just leave them alone; changes from `en.json` to the `en_*.json` files should propagate automatically, though it may take some time.

File: src/settings/Plugins/PluginForm.js [hunk 3]
```diff
@@ -31,21 +31,20 @@ class PluginForm extends React.Component {
...
-      module: '@@',
-      displayName: '(none)',
-    }].concat(pluginType).map(p => ({
-      value: p.module,
-      label: `${p.displayName}${p.version ? ` v${p.version}` : ''}`,
-    }));
 
+    const options = [{ value: '@@', label: '(none)' }].concat(pluginType.map(p => ({
+      value: p.module,
+      label: intl.formatMessage({ id: 'ui-tenant-settings.settings.pluginNames.' + p.pluginType }) + ' ' + p.version,
```
Comment (mkuklis) [inline src/settings/Plugins/PluginForm.js:39]: @K-Felk could you use string interpolation here?

**Comment by K-Felk** [comment]
> This looks great but there is a little bit of clean up work to do before you merge it.
> 
> Also note that changes to locale files other than `en.json` will be overwritten on the next import from Lokalise so I typically just leave them alone; changes from `en.json` to the `en_*.json` files should propagate automatically, though it may take some time.

I'm aware.  Because I have never used react-intl's API directly like this before, I felt it necessary to make sure the translation process would go off correctly-so I placed strings directly into the english-language files for testing purposes.  I didn't remove them because I know they'll eventually be overwritten anyway.

File: translations/ui-tenant-settings/en.json (hunk 2 above)
Comment (K-Felk) [reply translations/ui-tenant-settings/en.json:126]: I honestly wasn't sure what "finc" stood for-I'll expand them to "finance."

File: src/settings/Plugins/PluginForm.js (hunk 3 above)
Comment (K-Felk) [reply src/settings/Plugins/PluginForm.js]: I don't see why not.

**Pull request merged**